File: demo/java-demo/src/main/java/com/alibaba/demo/association/CookerService.java
package com.alibaba.demo.association;

/**
 * 目标类，此类中的一些调用将会被Mock掉
 * Target class, some invocations inside this class will be mocked
 */
public class CookerService {

    private static String hireSandwichCooker() {
        return "Real-Sandwich-Cooker";
    }

    private static String hireHamburgerCooker() {
        return "Real-Hamburger-Cooker";
    }

    private String cookSandwich() {
        return "Real-Sandwich";
    }

    private String cookHamburger() {
        return "Real-Hamburger";
    }

    public String prepareSandwich() {
        return hireSandwichCooker() + " & " + cookSandwich();
    }

    public String prepareHamburger() {
        return hireHamburgerCooker() + " & " + cookHamburger();
    }

}

File: demo/kotlin-demo/src/main/kotlin/com/alibaba/demo/association/CookerService.kt
package com.alibaba.demo.association

/**
 * 目标类，此类中的一些调用将会被Mock掉
 * Target class, some invocations inside this class will be mocked
 */
class CookerService {

    private fun cookSandwich(): String {
        return "Real-Sandwich"
    }

    private fun cookHamburger(): String {
        return "Real-Hamburger"
    }

    fun prepareSandwich(): String {
        return hireSandwichCooker() + " & " + cookSandwich()
    }

    fun prepareHamburger(): String {
        return hireHamburgerCooker() + " & " + cookHamburger()
    }

    companion object {
        private fun hireSandwichCooker(): String {
            return "Real-Sandwich-Cooker"
        }

        private fun hireHamburgerCooker(): String {
            return "Real-Hamburger-Cooker"
        }
    }
}
